### PR TITLE
Unit tests for the profile canister.

### PR DESCRIPTION
**Pull request by Nima-Ra**

### CODE REVIEW

File: profile/src/profile.rs [hunk 1]
```diff
@@ -217,3 +217,40 @@ fn set_profile(profile_data: ProfileMetadata) {
...
+        Principal::from_text("fterm-bydaq-aaaaa-aaaaa-c").unwrap()
+    }
+
+    #[test]
+    fn metadata() {
+        let mut profile_db = ProfileDB::default();
+
+        let mut barry_metadata: ProfileMetadata = ProfileMetadata { display_name: None, description: None, emoji: None, avatar: None, banner: None, version: 4 };
+
+        assert_eq!(profile_db.set_display_name(barry(), String::from("Barry Allen")), ());
```
[the remark this answers is not in the document]
Comment (Nima-Ra) [reply profile/src/profile.rs:236]: True. The Flash is the best lol

**Comment by Nima-Ra** [comment]
Updated the code to contain the partial and the null case tests.

1. If the requested profile doesn't exist, the expected return value is set to be `None`.

2. If only one of the fields is set (in this case the display name), the only field in the metadata that doesn't have a value of `None` would be the field we set and the version:

```
(
  opt record {
    banner = null;
    description = null;
    emoji = null;
    display_name = opt "Alec Holland";
    version = 0;
    avatar = null;
  },
)
```

**Review by kanej — CHANGES_REQUESTED** [review]
Can we split into three separate tests.

When a test fails, the name should indicate what the precise problem is to make it an effective regression suite (i.e. you change the default value, and that breaks the api).

**Comment by Nima-Ra** [comment]
Updated the code. Now each test case is separated from the others.

**Review by kanej — APPROVED** [review]
Thats brilliant. Lets merge this.

**Pull request merged**